libcxx/utils/libcxx/test/dsl.py

```diff
@@ -69,8 +69,12 @@ def f(config, *args, **kwargs):
             if cacheKey not in cache:
                 cache[cacheKey] = function(config, *args, **kwargs)
                 # Update the persistent cache so it knows about the new key
-                with open(persistentCache, "wb") as cacheFile:
+                # We write to a temporary file and rename the result to ensure
```

Comment:
Don't we have the same problem with the creation of the `.tmp` file now? Don't we need to generate a unique file name instead?

Reply:
Hmm that's a good point, in my limited testing this was enough to avoid the race, but really it should be a per-shard suffix. Will update to use the PID.